summingbird-storm/src/main/scala/com/twitter/summingbird/storm/SpoutProvider.scala

```diff
@@ -2,37 +2,35 @@ package com.twitter.summingbird.storm
 
 import backtype.storm.metric.api.IMetric
 import backtype.storm.task.TopologyContext
-import backtype.storm.topology.{IRichSpout, TopologyBuilder}
+import backtype.storm.topology.{ IRichSpout, TopologyBuilder }
 import com.twitter.algebird.Semigroup
 import com.twitter.algebird.util.summer.Incrementor
 import com.twitter.summingbird._
-import com.twitter.summingbird.batch.{BatchID, Timestamp}
-import com.twitter.summingbird.online.{Externalizer, executor}
-import com.twitter.summingbird.online.option.SummerBuilder
+import com.twitter.summingbird.batch.{ BatchID, Timestamp }
+import com.twitter.summingbird.online.{ Externalizer, executor }
+import com.twitter.summingbird.online.option.{ SourceParallelism, SummerBuilder }
 import com.twitter.summingbird.option.JobId
-import com.twitter.summingbird.planner.{Dag, SummerNode}
+import com.twitter.summingbird.planner.{ Dag, SummerNode }
 import com.twitter.summingbird.storm.planner.StormNode
 import com.twitter.tormenta.spout.Spout
 import com.twitter.summingbird.storm.Constants
 import com.twitter.summingbird.storm.option.SpoutStormMetrics
 import com.twitter.summingbird.storm.spout.KeyValueSpout
 
-/**
- * Created by pnaramsetti on 8/18/16.
- */
 case class SpoutProvider(storm: Storm, stormDag: Dag[Storm], node: StormNode, jobID: JobId)(implicit topologyBuilder: TopologyBuilder) {
 
   private def getOrElse[T <: AnyRef: Manifest](stormDag: Dag[Storm], queryNode: StormNode = node, default: T) = storm.getOrElse(stormDag, queryNode, default)
 
-  private var sourceParallelism = getOrElse(stormDag, node, Constants.DEFAULT_SOURCE_PARALLELISM).parHint
+  private def getSourceParallelism(sourceParOption: Option[SourceParallelism]) =
+    { getOrElse(stormDag, node, sourceParOption.getOrElse(Constants.DEFAULT_SOURCE_PARALLELISM)).parHint }
+
   private val metrics = getOrElse(stormDag, node, Constants.DEFAULT_SPOUT_STORM_METRICS)
 
-  private[storm] def getSpoutParallelism : Int = sourceParallelism
+  private def extractSourceAttributes(): (Spout[(Timestamp, Any)], Option[SourceParallelism]) =
```

Comment:
`fn()` is used for functions with side effects. This should just be: `def extractSourceAttributes: (..`